Update for Cake 5.x

Ridzuan94 opened the pull request:
Improvement for CakePHP5

Ridzuan94 committed:
Improve codes after rector PHP and CakePHP rules

Ridzuan94 committed:
Update composer.json to follow CakePHP5 skeleton

- Refer: https://github.com/cakephp/app/blob/5.x/composer.json

Ridzuan94 committed:
Run composer cs-fix

Ridzuan94 committed:
Delete containsNonAlphaNumeric validator method

- This validation method has been removed in CakePHP5.
- Refer: https://github.com/cakephp/validation/blob/5.x/Validator.php
- Refer: https://github.com/cakephp/validation/blob/e40b73b303e00e517e04dcceca2813944113081e/Validator.php#L1720-L1740

Ridzuan94 committed:
Update Validator.php

- Follow the same datatype argument from parent Validator CakePHP5.
- Refer: https://github.com/cakephp/validation/blob/5.x/Validator.php

- isArray() method has been replaced with array() method.
- Refer: https://book.cakephp.org/4/en/appendices/4-5-migration-guide.html#validation

Ridzuan94 committed:
Update tests/bootstrap.php

- To fix this issue when run composer test:
Error: Call to undefined function Tyrellsys\CakePHPValidator\Validation\__d()

Ridzuan94 committed:
Update cs-fix script at composer.json

Ridzuan94 committed:
Update README.md

Ridzuan94 committed:
Update ValidatorTest.php

- Fix based on the CakePHP5 Validator

Ridzuan94 commented on src/Validation/Validator.php:
```diff
@@ -4,717 +4,799 @@
-       return parent::hasAtMost($field, $count, $message, $when);
-   }
-
+    /**
+     * @inheritDoc
+     */
+    public function requirePresence(array|string $field, Closure|string|bool $mode = true, ?string $message = null): CakeValidator
+    {
+        $message = $message ?? __d('validation', 'requirePresence');
+
+        return parent::requirePresence($field, $mode, $message);
+    }
```
Fix indentation when running composer cs-fix.
Next, follow the argument data type based on the CakePHP5 Validator.
Refer: https://github.com/cakephp/validation/blob/5.x/Validator.php

Ridzuan94 commented on tests/TestCase/Validation/ValidatorTest.php:
```diff
@@ -114,7 +111,7 @@ public function testMessages()
             ->utf8Extended('column')
             ->integer('column')
-            ->isArray('column')
+            ->array('column')
```
`Validator::isArray()` is deprecated in CakePHP5. Need to use `Validator::array()` instead.
Refer: https://book.cakephp.org/4/en/appendices/4-5-migration-guide.html#validation

Ridzuan94 commented on tests/TestCase/Validation/ValidatorTest.php:
```diff
@@ -387,12 +383,7 @@ public function testEmptyXXXX()
-                'tmp_name' => 'tmp_name',
-                'error' => UPLOAD_ERR_NO_FILE,
-            ],
+            'file' => new UploadedFile('', 0, UPLOAD_ERR_NO_FILE, null, null),
```
The file attachment already changed from an array to an `UploadedFile` object.
Refer: https://book.cakephp.org/5/en/appendices/5-0-migration-guide.html#http

TABTyrell commented on tests/TestCase/Validation/ValidatorTest.php:
```diff
@@ -323,7 +321,6 @@ public function testNotScalar()
             'lessThan' => 'lessThan',
             'lessThanOrEqual' => 'lessThanOrEqual',
             'equals' => 'equals',
-            'notEquals' => 'notEquals',
```
For `notEquals`, the Validation::comparison() method uses the operator `!=`. 

Currently, the values being compared are `[]` and `1`. With these values, the comparison will always return true. Instead of removing notEquals, why not fix the following line:

https://github.com/Ridzuan94/cakephp-validator/blob/eb3ca1f6482e839e09001e1840939a95a592b7d2/tests/TestCase/Validation/ValidatorTest.php#L266

Ensure that the value being compared is consistent with what is used for validation in this line:

https://github.com/Ridzuan94/cakephp-validator/blob/eb3ca1f6482e839e09001e1840939a95a592b7d2/tests/TestCase/Validation/ValidatorTest.php#L312

This way, it will not be necessary to remove notEquals from the errors column. Please check.

Ridzuan94 replied:
@TABTyrell 
Understood. For this fixed at 12d461cf66b80d571777302a9540fcbb2162f721.
Already ran composer test and it worked as expected.
Thank you for the advice given appreciate it. 🙏

TABTyrell reviewed (changes requested):
@mtak3 I'm sorry, we are unable to set you as reviewer in this PR at the moment. Once Ridzuan has made the necessary fixes, I would like to request your review of the updated PR. We'll keep you informed.

Thank you for your understanding!

Ridzuan94 committed:
Update ValidatorTest::ValidatorTest()

- Set notEquals value to the correct one.
Refer comment :
https://github.com/tyrellsys/cakephp-validator/pull/1#discussion_r1919496830
